decawave_ble/configure_devices.py

```diff
@@ -26,6 +26,13 @@ def configure_devices_from_database(configuration_database):
         logger.info('Getting target data for {}'.format(target_device_name))
         target_data = configuration_database.get_target_data(target_device_name)
         logger.info('Target data: {}'.format(target_data))
+        network_id = None
+        if target_data.get('network_id_hex_string') is not None:
```

Comment:
we could be a little smarter about this I guess. if the datatype is an integer then process it as an int, but if it a string then test for that standard way of encoding numbers as strings, `0x0100`, `0b100000000`, and `256` are equivalent hex, binary, and decimal.

Reply:
Right, but what do we do if the input is `100`? Assume that it's decimal (since it contains no letters and there's no prefix like `0x` or `0b`)?

Reply:
Also, in the current implementation, whether the value arrives as a string or an integer is dependent on which database we're using (e.g., Pandas as one set of rules for guessing column types from a CSV file, I don't know what Honeycomb does), which is why I'm wondering if we need to push all of this type checking/conversion logic inside the `ConfigurationDatabase` class, so what emerges is predictable.

Reply:
yes. 100 would be decimal.

I think that using the 0x and 0b prefixes gives the predictability we want... in fact pandas may do the conversion for you if it encounters those encodings.

Reply:
OK. I'll implement as you suggest for now. Later perhaps we can revisit whether this should happen inside `ConfigurationDatabase`

Reply:
OK. How about [this](https://github.com/WildflowerSchools/decawave_ble/blob/b8b021c32477ae4d22be78c488c61a9238d0eacc/decawave_ble/configure_devices.py#L29-L36)?